### PR TITLE
add profile.ps1 creation for PowerShell Function Apps

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
When a PowerShell Function App is created, it will also be created with a profile.ps1 which is a script that will be executed at "cold start".

Initially, this profile.ps1 will automatically authenticate you to Azure if you have the Az.Accounts module available and also have setup MSI. The profile also has an example on how to enable the old AzureRm alias.

I didn't see any tests for the InitAction but if they are hiding somewhere, please let me know and I will gladly add tests.

cc @daxian-dbw @SteveL-MSFT @joeyaiello @anirudhgarg @paulbatum @ahmelsayed @pragnagopa

### CODE REVIEW

**Comment by ahmelsayed** [comment]
InitAction tests are [here](https://github.com/Azure/azure-functions-core-tools/blob/master/test/Azure.Functions.Cli.Tests/E2E/InitTests.cs)

Who calls `profile.ps1`? is that something the powershell worker does on launch?

**Comment by daxian-dbw** [comment]
@ahmelsayed Yes, powershell worker consumes the `profile.ps1` on launch to pre-configure the Runspace environment.

**Comment by TylerLeonhardt** [comment]
Thanks for the pointer to the InitAction tests! My vscode search was a tad strict.

**Comment by TylerLeonhardt** [comment]
@ahmelsayed any advice on how to run the tests? I tried running the E2E tests from the `master` branch on both macOS and Windows and got:

```
Failed   Azure.Functions.Cli.Tests.E2E.InitTests.init_with_worker_runtime(workerRuntime: "java")
Error Message:
 System.InvalidOperationException : Cannot start process because a file name has not been provided.
Stack Trace:
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Azure.Functions.Cli.Common.Executable.RunAsync(Action`1 outputCallback, Action`1 errorCallback, Nullable`1 timeout) in /Users/tyler/Code/CSharp/azure-functions-core-tools/src/Azure.Functions.Cli/Common/Executable.cs:line 55
   at Azure.Functions.Cli.Tests.E2E.Helpers.CliTester.InternalRun(String workingDir, RunConfiguration[] runConfigurations, ITestOutputHelper output, Boolean startHost) in /Users/tyler/Code/CSharp/azure-functions-core-tools/test/Azure.Functions.Cli.Tests/E2E/Helpers/CliTester.cs:line 84
   at Azure.Functions.Cli.Tests.E2E.Helpers.CliTester.Run(RunConfiguration[] runConfigurations, ITestOutputHelper output, String workingDir, Boolean startHost) in /Users/tyler/Code/CSharp/azure-functions-core-tools/test/Azure.Functions.Cli.Tests/E2E/Helpers/CliTester.cs:line 35
--- End of stack trace from previous location where exception was thrown ---
Standard Output Messages:
 stdout: Running: >  init . --worker-runtime java


Failed   Azure.Functions.Cli.Tests.E2E.InitTests.init_with_worker_runtime(workerRuntime: "node")
Error Message:
 System.InvalidOperationException : Cannot start process because a file name has not been provided.
Stack Trace:
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Azure.Functions.Cli.Common.Executable.RunAsync(Action`1 outputCallback, Action`1 errorCallback, Nullable`1 timeout) in /Users/tyler/Code/CSharp/azure-functions-core-tools/src/Azure.Functions.Cli/Common/Executable.cs:line 55
   at Azure.Functions.Cli.Tests.E2E.Helpers.CliTester.InternalRun(String workingDir, RunConfiguration[] runConfigurations, ITestOutputHelper output, Boolean startHost) in /Users/tyler/Code/CSharp/azure-functions-core-tools/test/Azure.Functions.Cli.Tests/E2E/Helpers/CliTester.cs:line 84
   at Azure.Functions.Cli.Tests.E2E.Helpers.CliTester.Run(RunConfiguration[] runConfigurations, ITestOutputHelper output, String workingDir, Boolean startHost) in /Users/tyler/Code/CSharp/azure-functions-core-tools/test/Azure.Functions.Cli.Tests/E2E/Helpers/CliTester.cs:line 35
--- End of stack trace from previous location where exception was thrown ---
Standard Output Messages:
 stdout: Running: >  init . --worker-runtime node
```

Naturally my test addition fails with the same error in my branch.

**Comment by ahmelsayed** [comment]
They need an env var `FUNC_PATH=/location/of/func`  to test. 

You can do

```
cd build
dotnet run Test
```

That should build it and run all the tests. See this comment for more details https://github.com/Azure/azure-functions-core-tools/pull/838#issuecomment-437171046 

I need to add the instructions to the readme still.

File: src/Azure.Functions.Cli/StaticResources/profile.ps1.template [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+# Authenticate with Azure PowerShell using MSI.
+# Remove this if you are not planning on using MSI or Azure PowerShell.
+if ($env:MSI_SECRET -and (Get-Module -ListAvailable Az.Accounts)) {
+    $tokenAuthURI = $env:MSI_ENDPOINT + "?resource=https://management.azure.com&api-version=2017-09-01"
+    $tokenResponse = Invoke-RestMethod -Method Get -Headers @{"Secret"="$env:MSI_SECRET"} -Uri $tokenAuthURI
+    Connect-AzAccount -AccessToken $tokenResponse.access_token -AccountId $env:WEBSITE_SITE_NAME
+}
+
+# Enable legacy AzureRm alias in Azure PowerShell.
```
Comment (SteveL-MSFT) [inline src/Azure.Functions.Cli/StaticResources/profile.ps1.template]:
```suggestion
# Uncomment the next line to enable legacy AzureRm alias in Azure PowerShell.
```

**Comment by eamonoreilly** [comment]
You might want to use Get-AzEnvironment to get the management URI in case the customer wants to use other endpoints for government / china clouds. 
Example: 
`
$Environment = "AzureCloud"

$AzureEnvironment  = Get-AzEnvironment -Name $Environment

$tokenAuthURI = $env:MSI_ENDPOINT + "?resource=$($AzureEnvironment.ResourceManagerUrl).&api-version=2017-09-01"
`

**Comment by anirudhgarg** [comment]
Just confirming this work has already been done on the PowerShell Worker and you are just replicating the behaviour inside the CLI ?

**Comment by daxian-dbw** [comment]
@anirudhgarg PowerShell worker will process a `profile.ps1` file if it's present in the Function App folder. The change here is to make `func init --worker-runtime powershell` create a default profile file.

> **this work** has already been done on the PowerShell Worker

If you are talking about authenticating to Azure, PowerShell worker no longer has that logic baked in. The default `powershell.ps1` is the way to go if a user wants this auto-authentication in place.

**Pull request merged**